utils/core/polyholes.scad

```diff
@@ -34,15 +34,15 @@
 //! leaving a scar on either surface.
 //
 function sides(r) = max(round(4 * r), 3);                                       //! Optimium number of sides for specified radius
-function corrected_radius(r, n = 0)   = r / cos(180 / (n ? n : sides(r)));      //! Adjusted radius to make flats lie on the circle
-function corrected_diameter(d, n = 0) = d / cos(180 / (n ? n : sides(d / 2)));  //! Adjusted diameter to make flats lie on the circle
+function corrected_radius(r, n = 0)   = r / cos(180 / (n == 0 ? r2sides(r) : n == 1 ? sides(r) : n));    //! Adjusted radius to make flats lie on the circle
+function corrected_diameter(d, n = 0) = corrected_radius(d / 2, n);                                     //! Adjusted diameter to make flats lie on the circle
 
-module poly_circle(r, sides = 0) { //! Make a circle adjusted to print the correct size
-    n = sides ? sides : sides(r);
-    circle(r = corrected_radius(r,n), $fn = n);
+module poly_circle(r, sides = 1) { //! Make a circle adjusted to print the correct size
+    n = sides == 1 ? sides(r) : sides;
```

Comment:
Doesn't like correct to me. n should either be sides(r) or r2sides(r)

Reply:
This has a facility for the user to force the number of sides, ie `poly_circle(5, 8)` forces an octagon. However, I think I have overlooked the n=0 case. I think the correction should be:
```
n = sides == 0 ? r2sides(r) : sides == 1 ? sides(r) : sides;
```
Do you agree?

Reply:
Yes that is correct but I think passing n to sides() and letting it sort out the number of sides will make the code a lot simpler. E.g.
```
function sides(r, n = undef) = is_undef(n) ? max(round(4 * r), 3) : n ? n : r2sides(r);                                       //! Optimium number of sides for specified radius or overridden by n which if 0 uses r2sides(r)
```

Then default `sides` to `undef` in all the clients and I think it will all get simpler. E.g.
```
function corrected_radius(r, n = undef)   = r / cos(180 / sides(r, n));      //! Adjusted radius to make flats lie on the circle

module poly_circle(r, sides = undef) { //! Make a circle adjusted to print the correct size
    n = sides(r, sides);
    circle(r = corrected_radius(r, n), $fn = n);
}
```
Etc.

Reply:
Do you need to explicitly assign `n =  undef``? Surely it is undefined if it is not assigned a value?

Reply:
I think its clearer that you can omit it and possibly OpenSCAD will warn if you don't pass enough parameters in the future as it gets more picky all the time.

Why did you not take my suggestion to handle it all in sides()?

I think poly_circle() doesn't work with sides = 0 in the current iteration.

Reply:
> Why did you not take my suggestion to handle it all in sides()?

I didn't see it. I'll make that change and resubmit.